### PR TITLE
replace mocha with jest

### PR DESCRIPTION
**Pull request by khassel**
As a first step of #2408 I managed to replace mocha with jest.

The new setup still uses spectron but we now can go the next steps as
- replace e2e test (see https://github.com/MichMich/MagicMirror/issues/2408#issuecomment-830678714) or
- find a replacement for spectron

### CODE REVIEW

File: modules/default/weatherforecast/weatherforecast.js [hunk 1]
```diff
@@ -339,7 +339,13 @@ Module.register("weatherforecast", {
...
 	 * argument data object - Weather information received form openweather.org.
 	 */
-	processWeather: function (data) {
+	processWeather: function (data, momenttz) {
+		let mom;
+		if (momenttz === null) {
+			mom = moment;
+		} else {
+			mom = momenttz;
+		}
```
Comment (rejas) [inline modules/default/weatherforecast/weatherforecast.js:348]: Could be shorter (and still readable) like this 
let mom = !momenttz ? moment : mom;

So we need this for the tests?

File: modules/default/weatherforecast/weatherforecast.js [hunk 2]
```diff
@@ -482,16 +488,22 @@ Module.register("weatherforecast", {
 	 * That object has a property "3h" which contains the amount of rain since the previous forecast in the list.
 	 * This code finds all forecasts that is for the same day and sums the amount of rain and returns that.
 	 */
-	processRain: function (forecast, allForecasts) {
+	processRain: function (forecast, allForecasts, momenttz) {
+		let mom;
```
Comment (rejas) [inline modules/default/weatherforecast/weatherforecast.js:492]: see comment above for shorter version

**Review by rejas — COMMENTED** [review]
great work! only a few coments :-)

File: modules/default/weatherforecast/weatherforecast.js (hunk 1 above)
Comment (MichMich) [reply modules/default/weatherforecast/weatherforecast.js:348]: In that case I'd prefer:
```
let mom = momenttz ? mom : moment; // Exception last.
```

I'm also not a big fan of abbreviations in variable names. Looking at the above code, I've got no idea what `mom` is.

File: tests/e2e/dev_console.js [hunk 3]
```diff
@@ -30,33 +21,35 @@ describe("Development console tests", function () {
...
-		after(function () {
-			return helpers.stopApplication(app);
-		});
-
-		it("should open dev console when provided", function () {
-			return expect(app.browserWindow.isDevToolsOpened()).to.eventually.equal(true);
-		});
-	});
-	/* eslint-enable */
+	// describe("With 'dev' commandline argument", function () {
```
[the remark this answers is not in the document]
Comment (khassel) [reply tests/e2e/dev_console.js:34]: no, I have this on my list to take a look later but I found no quick fix ...

File: js/logger.js [hunk 4]
```diff
@@ -32,10 +32,13 @@
 		groupCollapsed: Function.prototype.bind.call(console.groupCollapsed, console),
 		groupEnd: Function.prototype.bind.call(console.groupEnd, console),
 		time: Function.prototype.bind.call(console.time, console),
-		timeEnd: Function.prototype.bind.call(console.timeEnd, console),
-		timeStamp: Function.prototype.bind.call(console.timeStamp, console)
+		timeEnd: Function.prototype.bind.call(console.timeEnd, console)
 	};
 
+	if (process.env.NODE_ENV.trim() !== "test") {
```
[the remark this answers is not in the document]
Comment (khassel) [reply js/logger.js:39]: comment added

File: modules/default/weatherforecast/weatherforecast.js (hunk 1 above)
Comment (khassel) [reply modules/default/weatherforecast/weatherforecast.js]: short if added

File: modules/default/weatherforecast/weatherforecast.js (hunk 2 above)
Comment (khassel) [reply modules/default/weatherforecast/weatherforecast.js]: short if added

**Comment by MichMich** [comment]
Great work! Love it! :)

**Pull request merged**